src/fields/assembly_observe.hh

```diff
@@ -0,0 +1,235 @@
+/*!
+ *
+﻿ * Copyright (C) 2015 Technical University of Liberec.  All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 3 as published by the
+ * Free Software Foundation. (http://www.gnu.org/licenses/gpl-3.0.en.html)
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+ *
+ *
+ * @file    assembly_observe.hh
+ * @brief
+ */
+
+#ifndef ASSEMBLY_OBSERVE_HH_
+#define ASSEMBLY_OBSERVE_HH_
+
+#include <unordered_map>
+
+#include "coupling/generic_assembly.hh"
+#include "coupling/assembly_base.hh"
+#include "fem/dofhandler.hh"
+#include "fields/field_value_cache.hh"
+#include "io/observe.hh"
+
+
+/// Holds data of one eval point on patch (index of element and local coordinations).
+struct PatchPointData {
+    /// Default constructor
+    PatchPointData() {}
+
+    /// Constructor with data mebers initialization
+    PatchPointData(unsigned int elm_idx, arma::vec loc_coords)
+    : elem_idx(elm_idx), local_coords(loc_coords), i_quad(0), i_quad_point(0) {}
+
+    /// Copy constructor
+    PatchPointData(const PatchPointData &other)
```

Comment:
No constructors necessary for POD struct types.

Constructed simply by:
`PatchPointData x = {elm_idx, loc_coords, 0,0}`

or `emplace_back(idx, loc_coords, 0,0)`